src/api/app/controllers/webui/user_controller.rb

```diff
@@ -193,28 +193,36 @@ def change_password
       return
     end
 
-    # check the valid of the params
-    unless User.current.password_equals?(params[:password])
-      errmsg = 'The value of current password does not match your current password. Please enter the password and try again.'
-    end
-    if params[:new_password] != params[:repeat_password]
-      errmsg = 'The passwords do not match, please try again.'
-    end
-    if params[:password] == params[:new_password]
-      errmsg = 'The new password is the same as your current password. Please enter a new password.'
-    end
-    if errmsg
-      flash[:error] = errmsg
-      redirect_to action: :show, user: User.current
-      return
-    end
-
     user = User.current
-    user.update_password params[:new_password]
-    user.save!
 
-    flash[:success] = 'Your password has been changed successfully.'
-    redirect_to action: :show, user: User.current
+    if user.password_old
+      if user.password_old_equals?(params[:password])
+        user.password = params[:new_password]
+        user.password_old = nil
+        user.password_old_salt = nil
+        user.password_old_hash_type = nil
```

Comment:
What about using ``update``?
```user.update(password_old: nil, password_old_salt: nil, password_old_hash_type: nil)```
And btw: you're missing a save here :wink: 

Edit: Sorry, just saw the save in line 219. Then you could use ``assign_attributes``

Reply:
Hm, what about overwritting the ``authenticate`` method in the user model and doing all that stuff there? You can also use this function than in the model

Reply:
Also what you do here is basically the ``convert_password_function``, maybe you can reuse it here.

Reply:
Regarding the missing save: It's at line 219.